src/include/common/vector/auxiliary_buffer.h

```diff
@@ -56,6 +57,13 @@ class ArrowColumnAuxiliaryBuffer : public AuxiliaryBuffer {
     std::shared_ptr<arrow::Array> column;
 };
 
+class ArrowColumnChunkedAuxiliaryBuffer : public AuxiliaryBuffer {
```

Comment:
I don't think you should add a new class. Just reuse `ArrowColumnAuxiliaryBuffer`, and replace `arrow::Array` with `arrow::ChunkedArray`.